Source/Core/Core/HW/WiimoteEmu/Dynamics.cpp

```diff
@@ -87,178 +131,167 @@ void EmulateDynamicShake(NormalizedAccelData* const accel, DynamicData& dynamic_
-                         const bool upright)
+WiimoteCommon::DataReportBuilder::AccelData ConvertAccelData(const Common::Vec3& accel, u16 zero_g,
+                                                             u16 one_g)
 {
-  const ControllerEmu::Force::StateData swing = swing_group->GetState();
-
-  // Determine which axis is which direction
-  const std::array<int, 3> axis_map{{
-      upright ? (sideways ? 0 : 1) : 2,  // up/down
-      sideways,                          // left/right
-      upright ? 2 : (sideways ? 0 : 1),  // forward/backward
-  }};
-
-  // Some orientations have up as positive, some as negative
-  // same with forward
-  std::array<s8, 3> g_dir{{-1, -1, -1}};
-  if (sideways && !upright)
-    g_dir[axis_map[2]] *= -1;
-  if (!sideways && upright)
-    g_dir[axis_map[0]] *= -1;
-
-  for (std::size_t i = 0; i < swing.size(); ++i)
+  const auto scaled_accel = accel * (one_g - zero_g) / float(GRAVITY_ACCELERATION);
+
+  // 10-bit integers.
+  constexpr long MAX_VALUE = (1 << 10) - 1;
+
+  return {u16(MathUtil::Clamp(std::lround(scaled_accel.x + zero_g), 0l, MAX_VALUE)),
+          u16(MathUtil::Clamp(std::lround(scaled_accel.y + zero_g), 0l, MAX_VALUE)),
+          u16(MathUtil::Clamp(std::lround(scaled_accel.z + zero_g), 0l, MAX_VALUE))};
+}
+
+Common::Matrix44 EmulateCursorMovement(ControllerEmu::Cursor* ir_group)
+{
+  const auto cursor = ir_group->GetState(true);
+
+  using Common::Matrix33;
+  using Common::Matrix44;
+
+  // Values are optimized for default settings in "Super Mario Galaxy 2"
+  // This seems to be acceptable for a good number of games.
+  constexpr float YAW_ANGLE = 0.3f / 5 * 1.435f;
+  constexpr float PITCH_ANGLE = 0.3f / 5 * 1.34f;
+
+  // Nintendo recommends a distance of 1-3 meters.
+  constexpr float NEUTRAL_DISTANCE = 1.5;
+
+  constexpr float MOVE_DISTANCE = 1.0;
+
+  return Matrix44::Translate({0, MOVE_DISTANCE * float(cursor.z), 0}) *
+         Matrix44::FromMatrix33(Matrix33::RotateX(PITCH_ANGLE * cursor.y) *
+                                Matrix33::RotateZ(YAW_ANGLE * cursor.x)) *
+         Matrix44::Translate({0, -NEUTRAL_DISTANCE, 0});
+}
+
+void ApproachAngleWithAccel(RotationalState* state, const Common::Vec3& angle_target,
+                            float max_accel, float time_elapsed)
+{
+  const auto stop_distance =
+      Common::Vec3(CalculateStopDistance(state->angular_velocity.x, max_accel),
+                   CalculateStopDistance(state->angular_velocity.y, max_accel),
+                   CalculateStopDistance(state->angular_velocity.z, max_accel));
+
+  const auto offset = angle_target - state->angle;
+  const auto stop_offset = offset - stop_distance;
+
+  const Common::Vec3 accel{std::copysign(max_accel, stop_offset.x),
+                           std::copysign(max_accel, stop_offset.y),
+                           std::copysign(max_accel, stop_offset.z)};
+
+  // INFO_LOG(WIIMOTE, "ang.x: %f vel.x %f stop.x %f trgt.x %f offset.x %f accel.x %f",
```

Comment:
Do we want/need this Log for later, or should it be removed (being commented out code)?

Reply:
I'll remove them.